### ISSUE TITLE
[Feature Request] Resume Download

### ISSUE THREAD

**Issue by iopq** [futo-org/grayjay-android#835]
There should be a way to start the download again, if it failed. Right now I need to go find the video again (doesn't always find the video even with the exact title), select the quality I want, sound quality, and hit download.

Of course, once I do this, it will tell me that the video is already being downloaded, despite the download failing. So I have to remember to come back, cancel, and set the correct quality to try again

It would be better if I could resume the download from the point where it failed and just finish downloading it.

**Comment by bruceschaff** [comment]
Can I add manual pause and resume download to this request

**Comment by Kelvin-FUTO** [comment]
Downloads should automatically retry as long as you're connected to an unmetered connection like wifi (unless you turned this off in settings).

Killing the app during downloading will just restart the download next time you boot the app (and on wifi).
Currently we don't resume exactly where it left off and instead start over to avoid corruption, but we may add this in the future.

If a download fails it should automatically retry till succeeded. Unless its a critical exception that it cannot recover, in which case it would be best you give us that exception so we can resolve it.

In other words, this functionality should effectively already be built in but automatic. So if it does not do that, we'll need more info so we can resolve that.
